### PR TITLE
Add tests for controllers with rspec

### PR DESCRIPTION
**Pull request by hetrox8**
This branch's core goal is to: 
1. Write tests for controllers.
The main language used to achieve this include:
1.Rails 
2.Ruby

### CODE REVIEW

File: spec/request/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      get "/users/#{user.id}/posts/#{post.id}"
+
+      expect(assigns(:user)).to eq(user)
+      expect(assigns(:post)).to eq(post)
+      expect(assigns(:comments_count)).to eq(0)
+      expect(assigns(:likes_count)).to eq(0)
+    end
+
+    it 'renders the show template' do
```
Comment (mikethreels) [inline spec/request/posts_controller_spec.rb]: - [ ] According to the requirements the spec file should include tests that ensure the response body includes the correct placeholder text for Example `expect(response.body).to include("Welcome to user index page")`

File: spec/request/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      expect(assigns(:user)).to eq(user)
+    end
+
+    it 'renders the show template' do
+      user = User.create(name: 'John')
+
+      get "/users/#{user.id}"
+
+      expect(response).to render_template(:show)
+    end
```
Comment (mikethreels) [inline spec/request/users_controller_spec.rb]: - [ ] According to the requirements the spec file should include tests that ensure the response body includes the correct placeholder text for Example `expect(response.body).to include("Welcome to user index page")`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi @hetrox8, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job creating the spec files.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @hetrox8   👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
✅  Made suggested changes👍 
✅ Gitflow is followed. 👍 
✅ Professional README file. 👍 
✅ PR has a good title and summary. 👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**